analytics/loggable_objects.go

```diff
@@ -0,0 +1,96 @@
+package analytics
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"net/http"
+)
+
+type RequestType string
+
+const (
+	COOKIE_SYNC RequestType = "cookie_sync"
+	AUCTION     RequestType = "auction"
+	SETUID      RequestType = "set_uid"
+	AMP         RequestType = "amp"
+)
+
+//Loggable object of a transaction at /openrtb2/auction endpoint
+type AuctionObject struct {
+	Type      RequestType
+	Status    int
+	Error     []error
+	Request   openrtb.BidRequest
+	Response  openrtb.BidResponse
+	UserAgent string
+}
+
+type AmpObject struct {
+	Type      RequestType
+	Status    int
+	Error     []error
+	Request   string
```

Comment:
As a general rule, structured data is much better than strings. If you use strings, then each analytics module has to parse them. Besides making everyone duplicate work, it means that the Request/Response contracts are scattered across the project. The endpoints already need to be aware of the request/response format... so it's much better for them to do it.